src/annotator/anchoring/pdf.coffee

```diff
@@ -8,8 +8,11 @@ RenderingStates = require('../pdfjs-rendering-states')
 
 # Caches for performance
 
-# Map of page index to Promise<string>
+# Map of page index to page text content as a `Promise<string>`
 pageTextCache = {}
+# Two-dimensional map from `[quote][position]` to `{page, anchor}` intended to
+# optimize re-anchoring of a pair of quote and position selectors if the
+# position selector fails to anchor on its own.
```

Comment:
This is the kind of thing where I think using classes really helps code clarity. If you have a lot of code that just deals with objects (or dicts in Python) and arrays, but those object and arrays have certain fields and structure, but that structure isn't declared anywhere you just sort of have to look through all the code to see how things are used ... It's very hard to follow. Adding the comment here helps a lot.